server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -167,6 +156,12 @@ public class Node implements Closeable {
         Setting.Property.NodeScope
     );
 
+    public static final Setting<TimeValue> MAXIMUM_REINDEXING_TIMEOUT_SETTING = Setting.positiveTimeSetting(
```

Comment:
I think we might as well move these two settings closer to their use, i.e., to `ShutdownPrepareService`.

Reply:
Sure done.